R/HTMLBlock.R

```diff
@@ -0,0 +1,46 @@
+#' @title `HTMLBlock`
+#' @docType class
+#' @description
+#' Specialized `FileBlock` for managing HTML content in reports.
+#' It's designed to handle various HTML content, and render the report as HTML,
+#' however `htmlwidgets` objects can also be rendered to static document-ready format.
+#'
+#' @keywords internal
+HTMLBlock <- R6::R6Class( # nolint: object_name_linter.
+  classname = "HTMLBlock",
+  inherit = FileBlock,
+  public = list(
+    #' @description Initialize a `HTMLBlock` object.
+    #'
+    #' @param content An object that can be rendered as a HTML content assigned to
+    #'   this `HTMLBlock`
+    #'
+    #' @return Object of class `HTMLBlock`, invisibly.
+    initialize = function(content) {
+      if (!missing(content)) {
+        self$set_content(content)
+      }
+      invisible(self)
+    },
+    #' @description Sets content of this `HTMLBlock`.
+    #'
+    #' @param content An object that can be rendered as a HTML content
+    #' assigned to this `HTMLBlock`
+    #'
+    #' @return `self`, invisibly.
+    #' @examples
+    #' HTMLBlock <- getFromNamespace("HTMLBlock", "teal.reporter")
+    #' block <- HTMLBlock$new()
+    #' block$set_content(shiny::div("HTML Content"))
+    #'
+    set_content = function(content) {
```

Comment:
is this method really needed? You're saving an unprocessed content to the file. I think you can remove this method at all and inherit from `ContentBlock` instead.

Reply:
I think it is needed. The R6 classes are designed in a way that the `Block` classes set content which is consumed in two places:
1. The file is saved in temp directory and the name stored using the setter which is used to display the contents in the previewer using the getter `readRDS(<Block>$get_content())` or `knitr::image_uri(<Block>$get_content())`
2. Later this file is also used in the `Rmd` document to render the contents in the saved file.